config/ecs/ecs-cloudwatch.yaml

```diff
@@ -20,127 +20,53 @@ processors:
-        action: insert
-      - key: aws.ecs.task.pull_started_at
+      - key: aws.ecs.launchtype
         action: delete
-      - key: PullStoppedAt
-        from_attribute: aws.ecs.task.pull_stopped_at
-        action: insert
-      - key: aws.ecs.task.pull_stopped_at
+      - key: aws.ecs.cluster.arn
         action: delete
-      - key: AvailabilityZone
-        from_attribute: cloud.zone
-        action: insert
-      - key: cloud.zone
-        action: delete
-      - key: LaunchType
-        from_attribute: aws.ecs.task.launch_type
-        action: insert
-      - key: aws.ecs.task.launch_type
+      - key: cloud.provider
         action: delete
-      - key: Region
-        from_attribute: cloud.region
-        action: insert
-      - key: cloud.region
+      - key: cloud.platform
         action: delete
-      - key: AccountId
-        from_attribute: cloud.account.id
-        action: insert
       - key: cloud.account.id
         action: delete
-      - key: DockerId
-        from_attribute: container.id
-        action: insert
-      - key: container.id
-        action: delete
-      - key: ContainerName
-        from_attribute: container.name
-        action: insert
-      - key: container.name
-        action: delete
-      - key: Image
-        from_attribute: container.image.name
-        action: insert
-      - key: container.image.name
+      - key: cloud.region
         action: delete
-      - key: ImageId
-        from_attribute: aws.ecs.container.image.id
-        action: insert
-      - key: aws.ecs.container.image.id
+      - key: cloud.availability_zone
         action: delete
-      - key: ExitCode
-        from_attribute: aws.ecs.container.exit_code
-        action: insert
-      - key: aws.ecs.container.exit_code
+      - key: aws.log.group.names
         action: delete
-      - key: CreatedAt
-        from_attribute: aws.ecs.container.created_at
-        action: insert
-      - key: aws.ecs.container.created_at
+      - key: aws.log.group.arns
         action: delete
-      - key: StartedAt
-        from_attribute: aws.ecs.container.started_at
-        action: insert
-      - key: aws.ecs.container.started_at
+      - key: aws.log.stream.names
         action: delete
-      - key: FinishedAt
-        from_attribute: aws.ecs.container.finished_at
-        action: insert
-      - key: aws.ecs.container.finished_at
+      - key: host.image.id
         action: delete
-      - key: ImageTag
-        from_attribute: container.image.tag
-        action: insert
-      - key: container.image.tag
+      - key: host.name
         action: delete
+      - key: host.name
+        action: host.type
 
 exporters:
   awsemf/application:
     namespace: ECS/AWSOTel/Application
     log_group_name: '/aws/ecs/application/metrics'
     resource_to_telemetry_conversion:
       enabled: true
-    dimension_rollup_option: NoDimensionRollup
-    metric_declarations:
-      - dimensions: [ [ TaskDefinitionFamily ] ]
```

Comment:
I believe without this allow list, this will cause every resource dimension to be sent, which could be dozens, and likely fail to ingest to CW or otherwise cost too much. Don't we need to add a reasonable allow list here? I heard that TaskDefinitionFamily may not be sufficient, in that case what do we need and can we add them all here in the allow list instead of removing it?

Reply:
Yes @anuraaga. I am working on a new config. Discussing offline. Planning to allow a specific set of dimensions which are truly useful for ECS customers.